### PR TITLE
Add Docker image artifact URL to PR comment

### PR DESCRIPTION
**Pull request by kgal-pan**
## Status
Ready

## Related Content Pull Request
Related PR: link to the PR at demisto/content

## Related Issues
Related: https://jira-dc.paloaltonetworks.com/browse/CIAC-8620

## Description
In PRs originating from forked repos, post a message with the link to the generated Docker image in CircleCI artifacts and command to load it locally.

See https://github.com/demisto/dockerfiles/pull/23484#issuecomment-1981412985 for how the comment is supposed to be presented.

### CODE REVIEW

**Review by JudahSchwartz — COMMENTED** [review]
looks great! Minor comment, tag me when youre done testing for final review

File: docker/post_github_comment.py [hunk 1]
```diff
@@ -75,16 +77,29 @@ def main():
...
+            title +
+            "Docker automatic build at CircleCI has deployed your docker image: {}\n".format(args.docker_image) +
+            "It is available now on docker hub at: https://hub.docker.com/r/{}/tags\n".format(base_name) +
+            "Get started by pulling the image:\n" +
+            "```\n" +
+            "docker pull {}\n".format(args.docker_image) +
+            "```\n" +
+            docker_info
+        )
+    elif os.environ.get('CIRCLE_WORKFLOW_JOB_ID'):
```
[the remark this answers is not in the document]
Comment (kgal-pan) [reply docker/post_github_comment.py]: https://github.com/demisto/dockerfiles/pull/23484/commits/6910157e303e418d19758e98f63dea8c5efd0670#diff-60fefff01f17dfa99d578b4867e6b771976105619a11c5079c48cce67782b1a2R43-R50

File: docker/post_github_comment.py [hunk 2]
```diff
@@ -61,7 +103,7 @@ def main():
     inspect_format = f'''
 {{{{ range $env := .Config.Env }}}}{{{{ if eq $env "DEPRECATED_IMAGE=true" }}}}## 🔴 IMPORTANT: This image is deprecated 🔴{{{{ end }}}}{{{{ end }}}}
 ## Docker Metadata
-- Image Size: `{get_docker_image_size(args.docker_image)}`
+- Image Size: `{get_docker_image_size(args.docker_image, is_contribution=args.is_contribution)}`
```
Comment (kgal-pan) [inline docker/post_github_comment.py:106]: `is_contribution` arg was added to support getting the image size from the CircleCI artifacts when running in a contribution mode. If not, we get the error seen [here](https://app.circleci.com/pipelines/github/demisto/dockerfiles/83325/workflows/ee5becc4-861e-41b6-9e04-490cd39fff6a/jobs/89478/steps?invite=true#step-106-336):

```
[1] failed getting image size for image: devdemisto/bottle2:1.0.0.89478. Err: 404 Client Error: Not Found for url: https://hub.docker.com/v2/repositories/devdemisto/bottle2/tags/1.0.0.89478/
```

Comment (kgal-pan) [reply docker/post_github_comment.py:106]: The change result can be seen [here](https://app.circleci.com/pipelines/github/demisto/dockerfiles/83365/workflows/f30e83bf-0a4c-4024-a74c-0cfb490b8cc1/jobs/89518?invite=true#step-106-353) or https://github.com/demisto/dockerfiles/pull/23484#issuecomment-1981384890.

**Comment by kgal-pan** [comment]
@JudahSchwartz - Should be good to go. I added the following:
1. the env var as an arg as suggested in the [review](https://github.com/demisto/dockerfiles/pull/23484#discussion_r1514169210).
2. The preview of how the comment looks can be found in https://github.com/demisto/dockerfiles/pull/23484#issuecomment-1981384890. 
3. The comment now includes the Docker metadata as well. It was added only to prod comments (see https://github.com/demisto/dockerfiles/pull/23484#issuecomment-1981384890). This was a fix added in https://github.com/demisto/dockerfiles/pull/23484/commits/3caa1f229450f6ba536f3db656fac3f7e9e86bab.

This is the output when command is copied and run in local env:

```bash
❯ curl -L 'https://output.circle-artifacts.com/output/job/4dbef337-6798-4238-9a3c-53e2418d3090/artifacts/0/docker_images/devdemisto_bottle2:1.0.0.89518.tar.gz' | gunzip | docker load
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 24.9M  100 24.9M    0     0  4698k      0  0:00:05  0:00:05 --:--:-- 5971k
e6f9ed28d3c3: Loading layer [==================================================>]  2.048kB/2.048kB
3abcf15506eb: Loading layer [==================================================>]  731.6kB/731.6kB
Loaded image: devdemisto/bottle2:1.0.0.89518
```

File: docker/build_docker.sh [hunk 3]
```diff
@@ -353,7 +353,8 @@ function docker_build {
             IMAGESAVE=${ARTDIR}/$IMAGENAMESAVE
             docker save -o "$IMAGESAVE" ${image_full_name}
             gzip "$IMAGESAVE"
-            cat << EOF
+            ${DOCKER_SRC_DIR}/post_github_comment.py ${image_full_name} "--is_contribution"
+                        cat << EOF
 -------------------------
```
Comment (JudahSchwartz) [inline docker/build_docker.sh:358]:
```suggestion
            ${DOCKER_SRC_DIR}/post_github_comment.py ${image_full_name} "--is_contribution"
            cat << EOF
-------------------------
```

**Review by JudahSchwartz — APPROVED** [review]
fantastic! thanks for this change!!!

**Pull request merged**